src/models/__tests__/appmodel.test.ts

```diff
@@ -1,31 +1,31 @@
 import { AppModel } from "../appmodel";
 
-const appModel = new AppModel();
-
-describe("text transformation suite", () => {
-
-	it(`should make the textInput text lowercased
-		 when setLowercase method called`, () => {
-			 const initialText = "TEXT";
-			 const expectedResultText = "text";
-			 appModel.textInput = initialText;
-			 expect(appModel.textInput).toBe(initialText);
-
-			 appModel.setLowercase();
-
-			 expect(appModel.textInput).toBe(expectedResultText);
-		})
-
-	it(`should make the transformation text uppercased
-		 when setUppercase method called`, () => {
-			const initialText = "text";
-			const expectedResultText = "TEXT";
-			appModel.textInput = initialText;
-			expect(appModel.textInput).toBe(initialText);
-
-			appModel.setUppercase();
-
-			expect(appModel.textInput).toBe(expectedResultText);
-		})
-
-});
\ No newline at end of file
+describe("AppModel.setLowerCase", () => {
```

Comment:
Ideally, a unit test name should be something like:  [Function]_Should[DoSomething]_When[SomeCondition]

With the current configuration, test failure output looks like this:

```
AppModel.setLowerCase › should make the flavorInput text lowercased
expect(received).toBe(expected) // Object.is equality
   Expected: "texxt"
   Received: "text"
```

This is exactly how I want it to look.   Having just one describe() section for the entire class breaks this a bit and encourages grouping many tests into the same closure, which is also problematic. 

In practice, individual tests should be as independent as possible from other tests, including the objects they use.   Object re-use exposes tests to side effects that can make them flaky or incorrect.   

The other advantage of having data and objects local to a test is that the test is easier to read because it does both setup and evaluation in the same bit of code.  For this reason I also avoid writing testSetup methods and consider them a bad smell in the test code.

Reply:
I see
thanks Eric